File: langs/java/src/test/java/dev/dotlyte/DotlyteTest.java
package dev.dotlyte;

import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.*;

class DotlyteTest {

    @Test
    void loadWithDefaults() {
        Map<String, Object> defaults = new HashMap<>();
        defaults.put("port", 3000);
        defaults.put("debug", false);

        Config config = Dotlyte.load(
            LoadOptions.builder().defaults(defaults).build()
        );

        assertEquals(3000, config.get("port"));
        assertEquals(false, config.get("debug"));
    }

    @Test
    void loadEmpty() {
        Config config = Dotlyte.load(LoadOptions.builder().build());
        assertNotNull(config);
    }

    @Test
    void configGetNested() {
        Map<String, Object> db = new HashMap<>();
        db.put("host", "localhost");
        db.put("port", 5432);
        Map<String, Object> data = new HashMap<>();
        data.put("database", db);

        Config config = new Config(data);
        assertEquals("localhost", config.get("database.host"));
        assertEquals(5432, config.get("database.port"));
    }

    @Test
    void configGetDefault() {
        Config config = new Config(Map.of("existing", "value"));
        assertEquals("fallback", config.get("missing", "fallback"));
    }

    @Test
    void configRequireMissing() {
        Config config = new Config(Map.of());
        assertThrows(DotlyteException.class, () -> config.require("MISSING"));
    }

    @Test
    void configHas() {
        Config config = new Config(Map.of("port", 8080));
        assertTrue(config.has("port"));
        assertFalse(config.has("missing"));
    }
}

File: langs/php/tests/DotlyteTest.php
<?php

declare(strict_types=1);

namespace Dotlyte\Tests;

use Dotlyte\Config;
use Dotlyte\Dotlyte;
use Dotlyte\DotlyteException;
use Dotlyte\LoadOptions;
use PHPUnit\Framework\TestCase;

final class DotlyteTest extends TestCase
{
    public function testLoadReturnsConfig(): void
    {
        $config = Dotlyte::load(new LoadOptions(defaults: ['port' => 3000]));
        $this->assertInstanceOf(Config::class, $config);
    }

    public function testDefaults(): void
    {
        $config = Dotlyte::load(new LoadOptions(
            defaults: ['port' => 3000, 'debug' => false],
        ));
        $this->assertSame(3000, $config->get('port'));
        $this->assertFalse($config->get('debug'));
    }

    public function testDotNotationAccess(): void
    {
        $config = new Config([
            'port' => 8080,
            'database' => ['host' => 'localhost', 'port' => 5432],
        ]);
        $this->assertSame(8080, $config->port);
        $this->assertSame('localhost', $config->database->host);
    }

    public function testGetWithDefault(): void
    {
        $config = new Config([]);
        $this->assertSame('fallback', $config->get('missing', 'fallback'));
    }

    public function testRequireThrowsOnMissing(): void
    {
        $config = new Config([]);
        $this->expectException(DotlyteException::class);
        $config->require('MISSING');
    }

    public function testNestedGet(): void
    {
        $config = new Config([
            'database' => ['host' => 'localhost'],
        ]);
        $this->assertSame('localhost', $config->get('database.host'));
    }

    public function testHas(): void
    {
        $config = new Config(['port' => 8080]);
        $this->assertTrue($config->has('port'));
        $this->assertFalse($config->has('missing'));
    }

    public function testToArray(): void
    {
        $data = ['port' => 8080];
        $config = new Config($data);
        $this->assertSame($data, $config->toArray());
    }
}
